### PR TITLE
fix: Fairness metrics and tests

### PR DESCRIPTION
**Pull request by christinaexyou**
Updates the fairness metrics and tests by addressing the following issues:
* #42 
* #43 
* #44

## Summary by Sourcery

Improve fairness metric implementations and tests by integrating sklearn confusion matrix, refactoring test infrastructure to use synthetic data, and standardizing model interfaces.

Bug Fixes:
- Fix calculate_confusion_matrix to use sklearn.confusion_matrix for accurate tp, tn, fp, fn counts
- Correct parameter naming in IndividualConsistency and group metric calculators

Enhancements:
- Generate synthetic bank churn dataset for tests and eliminate reliance on static CSV and sample files
- Refactor test helper functions (train_model, truth_predict_output, get_privileged_unprivleged_split, get_labeled_data, get_processed_data) to accept parameters with defaults
- Annotate model arguments in individual and group fairness metric classes with ClassifierMixin

Tests:
- Update tests to rename prediction_provider to model and adapt calls in IndividualConsistency scenarios

Chores:
- Remove obsolete test data files (bank_churn_train.csv, input-sample.b64, output-sample.b64)

### CODE REVIEW

File: src/core/metrics/fairness/individual/individual_consistency.py [hunk 1]
```diff
@@ -15,21 +16,21 @@
...
         consistency =  1
         for sample in samples:
-            prediction_outputs = prediction_provider.predict(sample)
+            prediction_outputs = model.predict(sample)
             prediction_output = prediction_outputs[0]
             neighbors = proximity_function(sample, samples)
-            neighbors_outputs = prediction_provider.predict(neighbors)
+            neighbors_outputs = model.predict(neighbors)
             for output in prediction_outputs:
                 for neighbor_output in neighbors_outputs:
```
[the remark this answers is not in the document]
Comment (ruivieira) [reply src/core/metrics/fairness/individual/individual_consistency.py:34]: `prediction_output = prediction_outputs[0]` + `for output in prediction_outputs:`
Perhaps the expectation of the prediction_outputs could be changed in a separate PR.

File: src/core/metrics/fairness/individual/individual_consistency.py [hunk 2]
```diff
@@ -15,21 +16,21 @@ class IndividualConsistency:
...
-        :param samples a list of inputs to be tested for consistency
-        :param prediction_provider the model under inspection
+        :param samples: a list of inputs to be tested for consistency
+        :param model: the model under inspection
         return the consistency measure
         """
         consistency =  1
         for sample in samples:
-            prediction_outputs = prediction_provider.predict(sample)
+            prediction_outputs = model.predict(sample)
```
Comment (ruivieira) [inline src/core/metrics/fairness/individual/individual_consistency.py:30]: Could we add here a simple validation that the outputs are not empty?

**Review by ruivieira — APPROVED** [review]
